raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -2015,6 +2109,11 @@ private long pollUnattachedAsObserver(UnattachedState state, long currentTimeMs)
     }
 
     private long pollCurrentState(long currentTimeMs) throws IOException {
+        // TODO: File a jira for this: Okay for now but the logic should be a little different.
+        // Leader should only advance the log start offset if all of followers fetched past it or there is a timeout
+        // While followers should only call maybeUpdateLogStartOffset if the leader's logStartOffset is greater than ours
+        log.maybeUpdateLogStartOffset();
```

Comment:
This implementation is okay for now but this simple implementation should be improved by https://issues.apache.org/jira/browse/KAFKA-12155